pkg/logs/sender/batch_strategy.go

```diff
@@ -166,29 +168,64 @@ func (s *batchStrategy) flushBuffer(outputChan chan *message.Payload) {
 }
 
 func (s *batchStrategy) sendMessages(messages []*message.Message, outputChan chan *message.Payload) {
-	serializedMessage := s.serializer.Serialize(messages)
-	log.Debugf("Send messages for pipeline %s (msg_count:%d, content_size=%d, avg_msg_size=%.2f)", s.pipelineName, len(messages), len(serializedMessage), float64(len(serializedMessage))/float64(len(messages)))
+	var encodedPayload bytes.Buffer
+	compressor := s.compression.NewStreamCompressor(&encodedPayload)
+	if compressor == nil {
+		compressor = &compression.NoopStreamCompressor{Writer: &encodedPayload}
+	}
+
+	wc := newWriterWithCounter(compressor)
+
+	if err := s.serializer.Serialize(messages, wc); err != nil {
+		log.Warn("Encoding failed - dropping payload", err)
+		s.utilization.Stop()
+		return
+	}
 
-	encodedPayload, err := s.compression.Compress(serializedMessage)
-	if err != nil {
+	if err := compressor.Close(); err != nil {
 		log.Warn("Encoding failed - dropping payload", err)
 		s.utilization.Stop()
 		return
 	}
 
+	unencodedSize := wc.getWrittenBytes()
+	log.Debugf("Send messages for pipeline %s (msg_count:%d, content_size=%d, avg_msg_size=%.2f)", s.pipelineName, len(messages), unencodedSize, float64(unencodedSize)/float64(len(messages)))
```

Comment:
Is this log important ? Dropping it, or even dropping the content size/avg msg size from it would allow cleaning up the code a bit more

Reply:
Unfortunately, yes - I've seen this log referenced recently in an internal question about the batcher configuration so I think we should keep it around.